src/Curves/CurveOperations.jl

```diff
@@ -0,0 +1,226 @@
+    point[2] = center[2] + r*sin(theta*fctr)
+end
+
+
+function endPointsLineCurvePoints(xStart::Array{Float64}, xEnd::Array{Float64}, t::Array{Float64}, points::Array{Float64})
+    for i = 1:length(t)
+        points[i,1:2] = xStart[1:2] + t[i]*(xEnd[1:2] - xStart[1:2])
+    end
+end
+
+
+function endPointsLineCurvePoint(xStart::Array{Float64}, xEnd::Array{Float64}, t::Float64, point::Array{Float64})
+        point[1:2] = xStart[1:2] + t*(xEnd[1:2] - xStart[1:2])
+end
+
+
+function peEquationCurvePoints(xEqn, yEqn, t::Array{Float64}, points::Array{Float64,2})
+
+    argPart,eqString = keyAndValueFromString(xEqn)
+    xArgM = match(argRegex,argPart)
+    xArg  = Symbol(xArgM.match)
+    ex    = Meta.parse(eqString)
+
+    argPart,eqString = keyAndValueFromString(yEqn)
+    yArgM = match(argRegex,argPart)
+    yArg  = Symbol(yArgM.match)
+    ey    = Meta.parse(eqString)
+
+    for i = 1:length(t)
+        points[i,1] = evalWithDict(ex,Dict(xArg=> t[i]))
+        points[i,2] = evalWithDict(ey,Dict(yArg=> t[i]))
+    end
+end
+
+
+function  peEquationCurvePoint(xEqn, yEqn, t::Float64, point::Array{Float64})
+
+    argPart,eqString = keyAndValueFromString(xEqn)
+    xArgM = match(argRegex,argPart)
+    xArg  = Symbol(xArgM.match)
+    ex    = Meta.parse(eqString)
+
+    argPart,eqString = keyAndValueFromString(yEqn)
+    yArgM = match(argRegex,argPart)
+    yArg  = Symbol(yArgM.match)
+    ey    = Meta.parse(eqString)
+
+    point[1] = evalWithDict(ex, Dict(xArg=> t))
+    point[2] = evalWithDict(ey, Dict(yArg=> t))
+
+end
+
+
+function splineCurvePoints(nKnots::Int, splineData::Array{Float64,2}, points::Array{Float64,2})
+
+    xSpline = constructSpline(nKnots,splineData[:,1],splineData[:,2])
+    ySpline = constructSpline(nKnots,splineData[:,1],splineData[:,3])
+
+    sz   = size(points)
+    nPts = sz[1]
+    t = 0.0
+    for i = 1:nPts
+        t = (i-1)/(nPts-1)
+        points[i,1] = evalSpline(xSpline,t)
+        points[i,2] = evalSpline(ySpline,t)
+    end
+end
+
+
+function splineCurvePoint(nKnots::Int, splineData::Array{Float64,2}, t, point::Array{Float64})
+
+    xSpline = constructSpline(nKnots,splineData[:,1],splineData[:,2])
+    ySpline = constructSpline(nKnots,splineData[:,1],splineData[:,3])
+
+    point[1] = evalSpline(xSpline,t)
+    point[2] = evalSpline(ySpline,t)
+end
+
+
+# This function evaluates a string as an equation, might be redundant code
+# function parse_eval_dict(s::AbstractString, locals::Dict{Symbol})
+#     ex = Meta.parse(s)
+#     assignments = [:($sym = $val) for (sym,val) in locals]
+#     eval(:(let $(assignments...); $ex; end))
+# end
+
+
+function evalWithDict(ex, locals::Dict{Symbol})
+    assignments = [:($sym = $val) for (sym,val) in locals]
+    eval(:(let $(assignments...); $ex; end))
+end
+
+
+function curvePoints(crvDict::Dict{String,Any}, N::Int)
+# N = Number of intervals
+    curveType::String = crvDict["TYPE"]
+
+    if curveType == "PARAMETRIC_EQUATION_CURVE"
```

Comment:
Dispatch can only dispatch on types, not contents of variables. That is, one would have to store each curve as a different type in the parent `Dict` instead of a `Dict`. Then one can implement a `curvePoints(crvDict::MyCurveType, N)` for each curve type.

However, I recommend to leave this to a future PR and maybe record this in an issue as a potential improvement for more user extensibility.

Reply:
Another solution would be to use a Dict whose entries are Julia types (structs) which can then be dispatched on. However, I agree this should be for a future PR

Reply:
I add this to the meta-issue list in #16